evalml/tests/pipeline_tests/test_component_graph.py

```diff
@@ -230,7 +230,7 @@ def test_init_bad_graphs():
 def test_order_x_and_y():
     graph = {
         "Imputer": [Imputer, "X", "y"],
-        "OHE": [OneHotEncoder, "Imputer.x", "Imputer.y"],
+        "OHE": [OneHotEncoder, "Imputer.x", "y"],
```

Comment:
The new logic catches some cases where we created a graph and used a component's .y when it doesn't produce such!